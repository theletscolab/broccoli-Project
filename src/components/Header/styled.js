import styled from 'styled-components/macro'

export const HeaderStyled = styled.header`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.grey50};
  border-bottom: 2px solid ${({ theme }) => theme.colors.grey200};
  display: flex;
  height: 60px;
  position: fixed;
  top: 0;
  width: 100%;
`