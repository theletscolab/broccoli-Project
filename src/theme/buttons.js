export const buttonSizes = {
  small: {
    height: '42px',
    fontSize: '1.4rem',
    padding: '1.2rem 1.7rem'
  },
  medium: {
    height: '48px',
    fontSize: '1.6rem',
    padding: '1.4rem 2.4rem'
  },
  large: {
    height: '54px',
    fontSize: '1.8rem',
    padding: '1.6rem 2.8rem'
  }
}
