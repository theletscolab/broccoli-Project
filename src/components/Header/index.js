import React from 'react'
import Logo from '../Logo'
import { HeaderStyled } from './styled'

const Header = () => (
  <HeaderStyled>
    <Logo />
  </HeaderStyled>
)

export default Header
