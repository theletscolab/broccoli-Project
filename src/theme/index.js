import { buttonSizes } from './buttons'
// App theme

// Colors Palette
const awlxDarkBlue = '#612fff'
const awlxLightBlue = '#e9e3ff'
const white = '#ffffff'
const grey50 = '#fafafa'
const grey100 = '#f5f5f5'
const grey200 = '#eeeeee'
const grey300 = '#e0e0e0'
const grey400 = '#bdbdbd'
const grey500 = '#9e9e9e'
const grey600 = '#757575'
const grey700 = '#616161'
const grey800 = '#424242'
const grey900 = '#212121'
const black = '#000000'
const black15 = 'rgba(0, 0, 0, 0.15)'
const black25 = 'rgba(0, 0, 0, 0.25)'
const black50 = 'rgba(0, 0, 0, 0.50)'
const black75 = 'rgba(0, 0, 0, 0.75)'
const error = '#cc0000'
const bodyBg = white
const bodyColor = grey900

// Responsive breakpoints and media-queries (mobile first).
const breakpoints = ['576px', '768px', '992px', '1200px']

const fonts = {
  base: `'Montserrat', sans-serif`,
  caveat: `'Caveat', cursive`
}

const fontWeights = {
  light: 300,
  regular: 400,
  medium: 500,
  semiBold: 600,
  bold: 700
}

const headingFontFamily = fonts.base

const fontSizes = ['1.2rem', '1.6rem', '2rem', '2.4rem', '3.2rem', '4.8rem', '6.4rem']

const bodyText = fontSizes[1]
const headingWeight = fontWeights.bold
const headingLineHeight = 1.4

const theme = {
  breakpoints,
  mediaQueries: {
    sm: `@media screen and (min-width: ${breakpoints[0]})`,
    md: `@media screen and (min-width: ${breakpoints[1]})`,
    lg: `@media screen and (min-width: ${breakpoints[2]})`,
    xl: `@media screen and (min-width: ${breakpoints[3]})`
  },
  colors: {
    awlxDarkBlue,
    awlxLightBlue,
    white,
    grey50,
    grey100,
    grey200,
    grey300,
    grey400,
    grey500,
    grey600,
    grey700,
    grey800,
    grey900,
    black,
    black15,
    black25,
    black50,
    black75,
    bodyBg,
    bodyColor,
    error
  },
  space: [0, 4, 8, 16, 32, 64, 128, 256],
  fonts,
  fontSizes,
  bodyText,
  fontWeights,
  headingWeight,
  headingLineHeight,
  headingFontFamily,
  shadows: {
    small: '0 0 4px rgba(0, 0, 0, .125)',
    large: '0 0 24px rgba(0, 0, 0, .125)'
  },
  maxContainerWidth: '992px',
  buttonSizes
}

export default theme
