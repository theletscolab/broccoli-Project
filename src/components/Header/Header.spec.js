import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import Header from './index'

describe('components > Header', () => {
  let component
  describe('with defaults', () => {
    beforeAll(() => {
      component = render(<Header />, {
        wrapper: ({ children }) => <ThemeProvider theme={theme}> {children}</ThemeProvider>
      })
    })

    it('rendered without crashing', () => {
      expect(component.asFragment()).toMatchSnapshot()
    })
  })
})
