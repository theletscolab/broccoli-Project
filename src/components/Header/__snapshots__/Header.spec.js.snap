// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Header with defaults rendered without crashing 1`] = `
<DocumentFragment>
   
  <header
    class="styled__HeaderStyled-sc-cbvljd-0 bACyDG"
  >
    <a
      class="styled__StyledLogo-sc-1v84jjx-0 dfgzSu"
      href="/"
    >
      Broccoli & Co.
    </a>
  </header>
</DocumentFragment>
`;
